snuba/web/query.py

```diff
@@ -137,6 +137,60 @@ def _run_query_pipeline(
     )
 
 
+def _run_and_apply_column_names(
+    dataset: Dataset,
+    timer: Timer,
+    query_metadata: SnubaQueryMetadata,
+    from_date: datetime,
+    to_date: datetime,
+    referrer: str,
+    clickhouse_query: Query,
+    request_settings: RequestSettings,
+    reader: Reader[SqlQuery],
+) -> QueryResult:
+    """
+    Executes the query and, after that, replaces the column names in in
+    QueryResult with the names the user expects and that are stored in
+    the SelectedExpression objects in the Query.
+    This happens so that we can remove aliases from the Query AST since
+    those aliases now are needed to produce the names the user expects
+    in the output.
+    """
+
+    result = _format_storage_query_and_run(
+        dataset,
+        timer,
+        query_metadata,
+        from_date,
+        to_date,
+        referrer,
+        clickhouse_query,
+        request_settings,
+        reader,
+    )
+
+    name_alias_mappings = [
+        (select.name, select.expression.alias)
+        for select in clickhouse_query.get_selected_columns_from_ast()
+    ]
+    discrepancies = [
+        (name, alias) for name, alias in name_alias_mappings if name != alias
+    ]
+    if discrepancies:
+        logger.warning(
+            "Discrepancies between select clause names and aliases",
+            extra={"mappings": discrepancies},
+            exc_info=True,
+        )
+        return result
+
+    # TODO actually replace the column names in the result (data and
+    # meta) with the names the user expects from the SelectedExpression
+    # objects.
```

Comment:
Can you augment this comment to include that it isn't necessary to do this mapping _yet_ because we're not yet using internal aliases? I know it's temporary but it will help anyone who reads this in the interim and wonders just WTF is going on here.

Reply:
sure, will do.